sdk/core/azure-core/src/main/java/com/azure/core/models/PollResult.java

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.models;
+
+import com.azure.core.annotation.Immutable;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** Provides status details for long running operations. */
+@Immutable
+public final class PollResult {
```

Comment:
I noticed that we already have a `PollResponse<T>` and `PollResult` has a very similar name. Other name suggestions:
- PollDetails
- PollOperationDetails

cc: @JonathanGiles any thoughts on the name?

Reply:
I tend to agree - `PollResponse` and `PollResult` are too close to each other to be easily discerned by the end user. We should consider a better name. However, having said that, if we had some sample code demonstrating the common usage pattern for both `PollResponse` and `PollResult`, perhaps it would demonstrate that there is less confusion than anticipated.

Reply:
Scenario of `PollResult`:

It is used in LRO API, e.g. https://github.com/Azure/autorest.java/blob/main/typespec-tests/src/main/java/com/_specs_/azure/core/lro/standard/StandardClient.java
```java
SyncPoller<PollResult, User> beginCreateOrReplace(String name, User resource)

SyncPoller<PollResult, ExportedUser> beginExport(String name, String format)
```

It is the "result" of the "polling".

---

Scenario of `PollResponse<>`:

It is used in the handling of the LRO API, e.g.

```java
        SyncPoller<PollResult, User> poller = client.beginCreateOrReplace(
                "madge", new User("contributor"));

        PollResponse<PollResult> response = poller.waitForCompletion();
        Assertions.assertEquals(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED, response.getStatus());

        User user = poller.getFinalResult();
        Assertions.assertEquals("madge", user.getName());
```

It is akin to a `Response` of simple API. However it is not exactly same, as it have `T value`, but not `statusCode` or `headers` or `request`.

Reply:
Renamed to `PollOperationDetails`